spec/views/shared/champs/piece_justificative/_pj_link.html.haml_spec.rb

```diff
@@ -0,0 +1,51 @@
+require 'rails_helper'
+
+describe 'shared/champs/piece_justificative/_pj_link.html.haml', type: :view do
+  let(:champ) { create(:champ, :piece_justificative, :with_piece_justificative_file) }
+  let(:virus_scan) { nil }
+
+  before do
+    if virus_scan
+      champ.update(virus_scan: virus_scan)
+    end
+  end
+
+  subject { render 'shared/champs/piece_justificative/pj_link', champ: champ, user_can_upload: false }
+
+  context 'when there is no anti-virus scan yet' do
+    let(:virus_scan) { nil }
+
+    it 'allows to download the file' do
+      expect(subject).to have_link(champ.piece_justificative_file.filename.to_s)
+      expect(subject).to have_text('ce fichier n’a pas été analysé par notre antivirus')
+    end
+  end
+
+  context 'when the anti-virus scan is pending' do
+    let(:virus_scan) { create(:virus_scan, :pending) }
+
+    it 'displays the filename, but doesn’t allow to download the file' do
+      expect(subject).to have_text(champ.piece_justificative_file.filename.to_s)
+      expect(subject).not_to have_link(champ.piece_justificative_file.filename.to_s)
+      expect(subject).to have_text('analyse antivirus en cours')
+    end
+  end
```

Comment:
- when there is no anti-virus scan yet, it allows to download the file
- when the anti-virus scan is pending, it displays the filename, but doesn’t allow to download the file

Hors scope pour cette PR, mais est-ce que tu ne trouves pas ça surprenant comme comportement ?

Reply:
Hébin un peu. En fait l'intitulé de la spec que j'ai écrit est trompeur : ce n'est pas tant _"when there is no anti-virus scan yet"_ que _"when the antivirus is disabled"_. Je vais modifier ça.

Ce qui est curieux, c'est qu'on a pris une exception Sentry sur ce chemin (pas d'antivirus associé). Je pense que la PR tient toujours (elle corrige le crash) – mais par ailleurs je vais creuser les raisons pour lesquelles on a déclenché ce chemin.

Reply:
:bowing_man:

Reply:
:bowing_man: